celer/utils.pyx

```diff
@@ -4,28 +4,77 @@
 # License: BSD 3 clause
 
 cimport cython
-from scipy.linalg.cython_blas cimport ddot, dasum
+
+from scipy.linalg.cython_blas cimport ddot, dasum, daxpy, dnrm2, dcopy, dscal
+from scipy.linalg.cython_blas cimport sdot, sasum, saxpy, snrm2, scopy, sscal
+from scipy.linalg.cython_lapack cimport sposv, dposv
 from libc.math cimport fabs
+from cython cimport floating
+
+
+cdef floating fused_dot(int * n, floating * x, int * inc1, floating * y,
+                        int * inc2) nogil:
+    if floating is double:
+        return ddot(n, x, inc1, y, inc2)
+    else:
+        return sdot(n, x, inc1, y, inc2)
```

Comment:
I would prefer to do something like this:

https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/cd_fast.pyx#L169

as branch with an if for every function call will cause an overhead you can avoid.